src/blocks/net.rs

```diff
@@ -105,49 +309,99 @@ fn convert_speed(speed: u64) -> (f64, &'static str) {
 
 impl Block for Net {
     fn update(&mut self) -> Result<Option<Duration>> {
-        let current_rx = read_file(&format!("{}rx_bytes", self.device_path))?
-            .parse::<u64>()
-            .block_error("net", "failed to parse rx_bytes")?;
+        // Skip updating tx/rx if device is not up.
+        let is_up = try!(self.device.is_up());
+        if !is_up {
+            self.network.set_text("×".to_string());
+            self.output_tx.set_text("×".to_string());
+            self.output_rx.set_text("×".to_string());
+
+            return Ok(Some(self.update_interval));
+        } else {
+            self.network.set_text("".to_string());
+        }
+
+        // Update SSID and IP address every 30s.
+        let now = Instant::now();
+        if now.duration_since(self.last_update).as_secs() > 30 {
+            if let Some(ref mut widget) = self.ssid {
```

Comment:
The renaming from `self.ssid` -> `widget` and `device.ssid` -> `ssid` is kinda confusing to read. Same goes for the part below... Maybe you can find a more readable solution? I won't block the PR for this change though :)

Reply:
Naming should be clearer now.